docs/arcadia-users-group/20240416-intro-to-ml1/make_peptides_ml_dataset.md

```diff
@@ -0,0 +1,263 @@
+
+peptide_bioactivity_filtered_with_clusters <- left_join(clusters_filtered, peptide_bioactivity_filtered, by = c("member" = "idpeptide"))
+
+# filter to clusters that all have the same bioactivity for at least one bioactivity label
+# (e.g. all peptides have a predicted bioactivity of "antimicrobial" in a given cluster)
+peptide_bioactivity_filtered_with_clusters_long <- peptide_bioactivity_filtered_with_clusters %>%
+  pivot_longer(cols = starts_with("allergen"):last_col(), names_to = "bioactivity", values_to = "value")
+
+# Find clusters where all members have a 1 for any bioactivity
+clusters_all_1_any_bioactivity <- peptide_bioactivity_filtered_with_clusters_long %>%
+  group_by(rep, bioactivity) %>%
+  summarise(all_ones = all(value == 1), .groups = 'drop') %>%
+  filter(all_ones) %>%
+  distinct(rep, .keep_all = TRUE)
+
+head(clusters_all_1_any_bioactivity)
+
+peptide_group_bioactivities <- peptide_bioactivity_filtered_with_clusters %>%
+  filter(rep %in% clusters_all_1_any_bioactivity$rep) %>%
+  left_join(clusters_all_1_any_bioactivity, by = "rep") %>%
+  select(rep, member, member_sequence = sequence, group_bioactivity = bioactivity)
+
+# Record all bioactivity predictions for each peptide by concatenating the bioactivity names
+peptide_bioactivities_concatenated <- peptide_bioactivity_filtered_with_clusters_long %>%
+  filter(value == 1) %>%
+  group_by(member) %>%
+  summarise(all_bioactivities = str_c(bioactivity, collapse = ";"), .groups = 'drop')
+
+# Join this back to your original (or relevant) dataset to add the concatenated bioactivities column
+final_dataset <- peptide_group_bioactivities %>%
+  left_join(peptide_bioactivities_concatenated, by = "member") %>%
+  distinct()
+
+# join with other peptide metadata ----------------------------------------
+
+gene_ontology <- read_csv("~/Downloads/backup_peptipedia_29_03_2023/table_public.gene_ontology.csv",
+                          col_names = c("peptideid", "go_id", "go_description", "go_category")) %>%
+  select(peptideid, go_id)
+
+pfam <- read_csv("~/Downloads/backup_peptipedia_29_03_2023/table_public.pfam.csv",
+                 col_names = c("peptideid", "pfam_id", "pfam_name", "pfam_type"))
+
+final_dataset <- final_dataset %>%
+  left_join(gene_ontology, by = c("member" = "peptideid")) %>%
+  left_join(pfam, by = c("member" = "peptideid")) %>%
+  distinct()
+
+final_dataset %>% group_by(group_bioactivity) %>% tally()
+
+write_tsv(final_dataset, "~/Downloads/backup_peptipedia_29_03_2023/peptides_for_ml_tmp.tsv")
+```
+
+
+6. Convert the output TSV to a FASTA file that can be used to calculate peptide characteristics (molecular weight, etc).
+```
+cut -f2,3 peptides_for_ml_tmp.tsv | tail -n +2 | seqkit tab2fx > peptides_for_ml.fasta
+```
+
+7. Calculate peptide characteristics (length, mw, etc.)
+
+```
+curl https://raw.githubusercontent.com/Arcadia-Science/peptigate/main/scripts/characterize_peptides.py?token=GHSAT0AAAAAACPE43ZXXIARD4XPPPE2JMBWZPHWYPA
```

Comment:
Probably don't want to include the token (even if its soon to expire)

Reply:
edit: just remembered this repo is public 😬 so this token should be deleted asap.